lib/fluent/plugin/out_newrelic.rb

```diff
@@ -75,15 +75,13 @@ def configure(conf)
       end
 
       def package_record(record, timestamp)
-        if defined? timestamp.nsec
-          timestamp = timestamp * 1000 + timestamp.nsec / 1_000_000
-        end
         packaged = {
-          'timestamp' => timestamp,
```

Comment:
I guess it could be possible to use this here:
```
'timestamp' => resolveTimestamp(record['timestamp'], timestamp)
```

Nice thing to put the timestamp resolution logic in a separate method BTW 🎉